pkg/domain/infra/tunnel/containers.go

```diff
@@ -451,19 +451,19 @@ func (ic *ContainerEngine) ContainerExecDetached(ctx context.Context, nameOrID s
 }
 
 func startAndAttach(ic *ContainerEngine, name string, detachKeys *string, input, output, errput *os.File) error { //nolint
-	attachErr := make(chan error)
+	var attachErr error
 	attachReady := make(chan bool)
 	go func() {
-		err := containers.Attach(ic.ClientCxt, name, detachKeys, bindings.PFalse, bindings.PTrue, input, output, errput, attachReady)
-		attachErr <- err
+		attachErr = containers.Attach(ic.ClientCxt, name, detachKeys, bindings.PFalse, bindings.PTrue, input, output, errput, attachReady)
+		attachReady <- true
```

Comment:
This looks wrong. `attachReady` is passed to `containers.Attach(...)` so I think we're in a race.

Reply:
I think we actually don't guarantee that attachReady is set if containers.Attach exits with error. We could make that a guarantee of containers.Attach? Or maybe make attachErr a channel and do a select{} on attachReady, attachErr

Reply:
I guess we could set attachReady twice.